### PR TITLE
evmmax: Implement ecadd and ecmul bn254 precompiles

### PR DESCRIPTION
**Pull request by rodiazet**
- Implement `ecadd` and `ecmul` precompiles using evmmax API.
- Add unit tests.
- Enable tests in CI

### CODE REVIEW

**Review by pdobacz — COMMENTED** [review]
:+1: Just some suggestions to make conventions consistent with secp256k1 code

File: lib/evmone_precompiles/bn254.hpp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    0x30644e72e131a029b85045b68181585d97816a916871ca8d3c208c16d87cfd47_u256;
+
+namespace evmmax::bn254
+{
+using Point = ecc::Point<uint256>;
+
+bool validate(const Point& pt) noexcept;
+uint256 field_inv(const ModArith<uint256>& m, const uint256& x) noexcept;
+
+Point bn254_add(const Point& pt1, const Point& pt2) noexcept;
```
[the remark this answers is not in the document]
Comment (rodiazet) [reply lib/evmone_precompiles/bn254.hpp]: Done. I made more changes to make it similar to secp256k1 code.

File: lib/evmone_precompiles/bn254.cpp [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+{
+    if (pt1.is_inf())
+        return pt2;
+    if (pt2.is_inf())
+        return pt1;
+
+    const evmmax::ModArith s{BN254Mod};
+
+    // b3 == 9 for y^2 == x^3 + 3
+    const auto b3 = s.to_mont(9);
```
[the remark this answers is not in the document]
Comment (rodiazet) [reply lib/evmone_precompiles/bn254.cpp]: done

File: lib/evmone_precompiles/bn254.cpp [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+
+    const auto pr = ecc::mul(s, ecc::to_proj(s, pt), c, b3);
+
+    return ecc::to_affine(s, field_inv, pr);
+}
+
+uint256 field_inv(const ModArith<uint256>& m, const uint256& x) noexcept
+{
+    // Inversion computation is derived from the addition chain:
```
[the remark this answers is not in the document]
Comment (rodiazet) [reply lib/evmone_precompiles/bn254.cpp:68]: done

**Review by pdobacz — APPROVED** [review]
Thank you for the updates, LGTM but would suggest another pair of eyes here

**Review by chfast — APPROVED** [review]

**Pull request merged**